File: eco-drive-symfony/src/Document/Incident.php
<?php

namespace App\Document;

use Doctrine\ODM\MongoDB\Mapping\Annotations as ODM;

#[ODM\EmbeddedDocument]
class Incident
{
    #[ODM\Field(type: "string")]
    private string $date;

    #[ODM\Field(type: "string")]
    private string $type;

    #[ODM\Field(type: "string")]
    private string $description;

    public function __construct(
        string $date = "",
        string $type = "",
        string $description = ""
    ) {
        $this->date = $date;
        $this->type = $type;
        $this->description = $description;
    }

    public function getDate(): string { return $this->date; }
    public function setDate(string $date): self { $this->date = $date; return $this; }

    public function getType(): string { return $this->type; }
    public function setType(string $type): self { $this->type = $type; return $this; }

    public function getDescription(): string { return $this->description; }
    public function setDescription(string $description): self { $this->description = $description; return $this; }
}

File: spring-boot-starter-data-rest/src/main/java/fr/dawan/formation/models/Incident.java
package fr.dawan.formation.models;

public class Incident {
    private String date;
    private String type;
    private String description;

    public Incident() {
    }

    public Incident(String date, String type, String description) {
        this.date = date;
        this.type = type;
        this.description = description;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "Incident{" +
            "date='" + date + '\'' +
            ", type='" + type + '\'' +
            ", description='" + description + '\'' +
            '}';
    }
}